operator/v1/types_ingress.go

```diff
@@ -350,6 +357,124 @@ const (
 	StrictNamespaceOwnershipCheck NamespaceOwnershipCheck = "Strict"
 )
 
+// LoggingDestinationType is a type of destination to which to send log
+// messages.
+//
+// +kubebuilder:validation:Enum=Container;Syslog
+type LoggingDestinationType string
+
+const (
+	// Container sends log messages to a sidecar container.
+	ContainerLoggingDestinationType LoggingDestinationType = "Container"
+
+	// Syslog sends log messages to a syslog endpoint.
+	SyslogLoggingDestinationType LoggingDestinationType = "Syslog"
+
+	// ContainerLoggingSidecarContainerName is the name of the container
+	// with the log output in an ingress controller pod when container
+	// logging is used.
+	ContainerLoggingSidecarContainerName = "logs"
+)
+
+// SyslogLoggingDestinationParameters describes parameters for the Syslog
+// logging destination type.
+type SyslogLoggingDestinationParameters struct {
+	// endpoint identifies the syslog endpoint that receives log messages.
+	// The endpoint must be a host name or IP address and a UDP port number.
+	// Specify the value in the format of a colon-separated name or address
+	// and port number: host:port
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
```

Comment:
Is there a better static validation for this? MinLength seems basically pointless now if the format is host:port.

Reply:
Maybe just keep required and remove MinLength?